### DISCUSSION TITLE
Deno run like a service

### DISCUSSION THREAD

**Discussion by ghost** [denoland/deno#18081] [General]
Why isn't Deno able to run a script in the background? 

Yes, I know PM2, systemetc, service etc.... 

But: 
+ for PM2 I need npm, 
+ for systemetc and service I need elevated rights like sudo, i.e. root rights. 

If I want to run Deno as a service on a web space, I have to assume that the options mentioned above may not be available to me. 
So - why can't Deno do it himself?

**Comment by Hexagon** [comment]
Check out https://github.com/hexagon/pup , early stage - but working. Manual at https://hexagon.github.io/pup .

**Reply by ghost** [reply]
https://github.com/Hexagon/pup/issues/9

**Comment by ghost** [comment]
Hey, I've tested pup and am amazed by its ease of launching multiple processes and configuring them. But - unfortunately - I had hoped to be able to do without the "screen" program, which unfortunately is not the case. However, the peculiarity of a created server is that if you press CTRL+C, the server will also be stopped without using "screen". So now I only use one screen in the background and let pup manage the other servers. That works too. In terms of the number of processes, I'm definitely half as much as if I create a "screen" session for each server, and I save about 200MB of RAM on the contrary. So yes - pup is a great thing - thanks for the work.
Who are you financially supported by, or what is your motivation to get pup suf up and take care of it?

**Reply by Hexagon** [reply]
Great to hear you have use for it  👍 

Pup is an idea i got about a week ago, when i noticed there is no "native" process manager for pup, I'm not financially supported at all, i just think making great software it's great recreation.

To get rid of screen you'll have to add pup to systemd or a similar service manager. I'll look into adding this to the documentation. Personally i use pup through docker, to get automatic start at boot etc. https://hexagon.github.io/pup/examples/docker/README.html

**Reply by Hexagon** [reply]
@ch-devhost Added instructions on how to run pup using docker, systemd user mode (non privileged) and launch user mode (non privileged). This could help you get rid of screen.

https://hexagon.github.io/pup/service.html
